src/App.tsx

```diff
@@ -1,27 +1,91 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { getAll, get5First, getRedGoods } from './api/goods';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [handleError, setError] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadAll = async () => {
+    setError(false);
+    try {
+      const allGoods = await getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      setGoods(allGoods);
+    } catch {
+      setError(true);
+    }
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const loadFive = async () => {
+    setError(false);
+    try {
+      const firstFiveGoods = await get5First();
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      setGoods(firstFiveGoods);
+    } catch {
+      setError(true);
+    }
+  };
+
+  const loadRed = async () => {
+    setError(false);
+    try {
+      const RedGoods = await getRedGoods();
+
+      setGoods(RedGoods);
+    } catch {
+      setError(true);
+    }
+  };
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={loadAll}
+      >
+        Load all goods
+      </button>
+
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={() => loadFive()}
```

Comment:
```suggestion
        onClick={loadFive}
```
use shorter variant everywhere